src/app/network/UploadManager.ts

```diff
@@ -101,6 +102,41 @@ class UploadManager {
     (fileData, next: (err: Error | null, res?: DriveFileData) => void) => {
       if (this.abortController?.signal.aborted ?? fileData.abortController?.signal.aborted) return;
 
+      if (window.performance && (window.performance as any).memory) {
+        const memory = (window.performance as unknown as { 
```

Comment:
To avoid having to make these castings, we can overwrite the Performance interface by adding this to the globals.d.ts file:
```
interface Performance extends Performance {
  memory?: {
    jsHeapSizeLimit: number;
    totalJSHeapSize: number;
    usedJSHeapSize: number;
  };
}
```

This way we already have the typing added in case we want to use this API in any other part of the application.